resilience4j-rxjava2/src/main/java/io/github/resilience4j/bulkhead/operator/DisposableBulkhead.java

```diff
@@ -25,22 +24,53 @@
 
     @Override
     public void dispose() {
-        if (disposed.compareAndSet(false, true)) {
+        if (DisposableHelper.dispose(this)) {
             releaseBulkhead();
-            disposable.dispose();
         }
     }
 
     @Override
     public boolean isDisposed() {
-        return disposed.get();
+        return DisposableHelper.isDisposed(get());
     }
 
-    protected void setDisposable(Disposable disposable) {
-        this.disposable = requireNonNull(disposable);
+    protected void onSubscribe(Disposable disposable, Consumer<Disposable> onSubscribe, Consumer<Throwable> onError) {
+        DisposableHelper.setOnce(this, disposable);
```

Comment:
You only need this if cancellation can happen before or during `onSubscribe`.

Reply:
I'm not sure if that can be ensured here. We can keep it just to be safe.